### PR TITLE
util: fix memory.reArrangeFallback cpu usage

### PR DESCRIPTION
**Pull request by wshwsh12**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/30353

Problem Summary:

### What is changed and how it works?

In my machine, this pr reduce execution time of the sql.
Before this pr:  2.5s+
After this pr: 0.3~0.4s

Profiles:
image
reArrangeFallback will not use too many cpu

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility 

### Release note

```release-note
Fix the problem of high use of reArrangeFallback cpu.
```

### CODE REVIEW

**Comment by ichn-hu** [comment]
will this fix help with https://github.com/pingcap/tidb/issues/31060 ?

File: store/copr/coprocessor.go [hunk 1]
```diff
@@ -1313,6 +1313,7 @@ func (e *rateLimitAction) close() {
 	e.conditionLock()
 	defer e.conditionUnlock()
 	e.cond.exceeded = false
+	e.SetFinished()
```
Comment (guo-shaoge) [inline store/copr/coprocessor.go:1316]: Just a question, this is where makes the sql is slow in #30353 ?

**Review by guo-shaoge — APPROVED** [review]

**Review by fzhedu — APPROVED** [review]
LGTM

**Pull request merged**